libkbfs/folder_branch_ops.go

```diff
@@ -4949,7 +4953,9 @@ func (fbo *folderBranchOps) registerAndWaitForUpdates() {
 		// Never give up hope until we shut down
 		expBackoff.MaxElapsedTime = 0
 		// Register and wait in a loop unless we hit an unrecoverable error
-		ctx, cancel := context.WithCancel(ctx)
+		fbo.cancelUpdatesLock.Lock()
+		ctx, fbo.cancelUpdates = context.WithCancel(ctx)
```

Comment:
I can do that, and will, but I think it's impossible to get here a second time without it having been called once already.